addon/components/materialize-carousel.js

```diff
@@ -0,0 +1,32 @@
+import Ember from 'ember';
+import layout from '../templates/components/materialize-carousel';
+
+export default Ember.Component.extend({
+  layout,
+  classNames: ['carousel'],
+
+  didUpdateAttrs (attrs) {
+    this._super (...arguments);
+    if (Ember.isPresent (attrs.newAttrs.fullWidth) || Ember.isPresent (attrs.newAttrs.duration) ||
+      Ember.isPresent (attrs.newAttrs.dist) || Ember.isPresent (attrs.newAttrs.shift) ||
+      Ember.isPresent (attrs.newAttrs.padding) || Ember.isPresent (attrs.newAttrs.indicators) ||
+      Ember.isPresent (attrs.newAttrs.noWrap)) {
+      this._init ();
```

Comment:
@ev-wilt Unfortunately, this will not give the wanted behavior. The reason is that the parent element is not guaranteed to be in the DOM when this method is called. Please see the documentation on [component lifecycle](https://guides.emberjs.com/v2.14.0/components/the-component-lifecycle/). You cannot call `carousel()` until `didInsertElement()` is called on the component. Only then is it guaranteed that component has been added to the DOM.

If you want to handle the case with the component is first rendered and then re-rendered, you may want to move all your logic to `didRender()`. This will be called last when the component is (re)rendered.